pkg/services/ssosettings/strategies/oauth_strategy.go

```diff
@@ -29,46 +30,64 @@ func (s *OAuthStrategy) IsMatch(provider string) bool {
-		"allowed_groups":             section.Key("allowed_groups").Value(),
+	result := &social.OAuthInfo{
+		AllowAssignGrafanaAdmin: social.MustBool(section.Key("allow_assign_grafana_admin").Value(), false),
+		AllowSignup:             social.MustBool(section.Key("allow_sign_up").Value(), false),
+		AllowedDomains:          util.SplitString(section.Key("allowed_domains").Value()),
+		AllowedGroups:           util.SplitString(section.Key("allowed_groups").Value()),
+		ApiUrl:                  section.Key("api_url").Value(),
+		AuthStyle:               section.Key("auth_style").Value(),
+		AuthUrl:                 section.Key("auth_url").Value(),
+		AutoLogin:               social.MustBool(section.Key("auto_login").Value(), false),
+		ClientId:                section.Key("client_id").Value(),
+		ClientSecret:            section.Key("client_secret").Value(),
+		EmailAttributeName:      section.Key("email_attribute_name").Value(),
+		EmailAttributePath:      section.Key("email_attribute_path").Value(),
+		EmptyScopes:             social.MustBool(section.Key("empty_scopes").Value(), false),
+		Enabled:                 social.MustBool(section.Key("enabled").Value(), false),
+		GroupsAttributePath:     section.Key("groups_attribute_path").Value(),
+		HostedDomain:            section.Key("hosted_domain").Value(),
+		Icon:                    section.Key("icon").Value(),
+		Name:                    section.Key("name").Value(),
+		RoleAttributePath:       section.Key("role_attribute_path").Value(),
+		RoleAttributeStrict:     social.MustBool(section.Key("role_attribute_strict").Value(), false),
+		Scopes:                  util.SplitString(section.Key("scopes").Value()),
+		SignoutRedirectUrl:      section.Key("signout_redirect_url").Value(),
+		SkipOrgRoleSync:         social.MustBool(section.Key("skip_org_role_sync").Value(), false),
+		TeamIdsAttributePath:    section.Key("team_ids_attribute_path").Value(),
+		TeamsUrl:                section.Key("teams_url").Value(),
+		TlsClientCa:             section.Key("tls_client_ca").Value(),
+		TlsClientCert:           section.Key("tls_client_cert").Value(),
+		TlsClientKey:            section.Key("tls_client_key").Value(),
+		TlsSkipVerify:           social.MustBool(section.Key("tls_skip_verify_insecure").Value(), false),
+		TokenUrl:                section.Key("token_url").Value(),
+		UsePKCE:                 social.MustBool(section.Key("use_pkce").Value(), false),
+		UseRefreshToken:         social.MustBool(section.Key("use_refresh_token").Value(), false),
+		Extra:                   map[string]string{},
 	}
 
-	// when empty_scopes parameter exists and is true, overwrite scope with empty value
-	if section.Key("empty_scopes").MustBool(false) {
-		result["scopes"] = []string{}
+	extraFields := getExtraKeysForProvider(provider)
+	for _, key := range extraFields {
+		result.Extra[key] = section.Key(key).Value()
 	}
+
 	return result, nil
 }
+
+func getExtraKeysForProvider(provider string) []string {
```

Comment:
You could replace this function by a map.

Reply:
💯 Thanks!